### ISSUE TITLE
[java] TestClassWithoutTestCases should not flag classes that begin with "Test"

### ISSUE THREAD

**Issue by jbisotti** [pmd/pmd#4382]
**Affects PMD Version:** 
6.51.0

**Rule:**
TestClassWithoutTestCases

<https://pmd.github.io/latest/pmd_rules_java_errorprone.html#testclasswithouttestcases>

**Description:**
A production class named `TestFlagFinder` is erroneously flagged as a violation.

```shell
[INFO] PMD Failure: com.foo.TestFlagFinder:32 Rule:TestClassWithoutTestCases Priority:3 The class 'TestFlagFinder' might be a test class, but it contains no test cases..
```

**Code Sample demonstrating the issue:**

```java
public class TestFlagFinder  {
}
```

**Expected outcome:**

This class should not be flagged as a violation of this rule.

**Running PMD through:** *Maven*

**Comment by jsotuyod** [comment]
This rule applies heuristics to detect classes that should be tests, but are not. Part of that is checking the classname against a regular expression, which defaults to `^(?:.*\.)?Test[^\.]*$|^(?:.*\.)?.*Tests?$|^(?:.*\.)?.*TestCase$`. As stated in the documentation, this is exposed under a property for customization. I don't think there is a clear issue with the default itself. It seems that, for this particular project, you will want to change this property to avoid matching this class.

**Comment by embee1981** [comment]
I think there is a small problem with the docs for this rule as I just hit the same problem with a class that starts with "Test" but is not a test class when I upgraded. The docs state 

> Test classes typically end with the **suffix** "Test", "Tests" or "TestCase". Having a non-test class with that name is not a good practice, since most people will assume it is a test case. Test classes have test methods named "testXXX" (JUnit3) or use annotations (e.g. @Test).

> The **suffix** can be configured using the property testClassPattern. To disable the detection of possible test classes by name, set this property to an empty string.

implying only suffixes are checked, not prefixes.
